experiments/scripts/sweep_checkpoints.py

```diff
@@ -0,0 +1,197 @@
+
+
+def cleanup_checkpoints(path):
+    """ 
+    Given a path, recursively search all folders for all folders
+    with one or more checkpoints in it. In each of those folders,
+    rename the highest numbered checkpoint to be 0.ckpt and delete
+    all others
+    """
+    for dirpath, _, filenames in os.walk(path):
+        cleanup = False
+        for filename in filenames:
+            if filename.endswith(".ckpt"):
+                cleanup = True
+                break
+        if cleanup:
+            filenames.sort()
+            os.rename(f"{dirpath}/{filenames[-1]}", f"{dirpath}/zzz")
+            for filename in filenames[:-1]:
+                os.remove(f"{dirpath}/{filename}")
+            os.rename(f"{dirpath}/zzz", f"{dirpath}/0.ckpt")
+
+
+def parse_experiment_output(output):
+    """
+    Returns a dict of key values from an experiment output
+    """
+    ret = {}
+
+    output_str = output.decode('utf-8')
+    lines = output_str.split("\n")
+    for line in lines:
+        if re.search(':', line):
+            pair = line.split(":", 1)
+            if pair[0] != "WARNING":
+                ret[pair[0]] = pair[1].lstrip()
+
+    return ret
+
+
+def run_all_models():
+    percentiles = []
+    below_cutoff = []
+    num_measurements = []
+    too_many_measurements = []
+    too_few_measurements = []
+
+    PERCENTILE_CUTOFF = 0.70
+    NUM_MEASUREMENTS_MAX = 15
+    NUM_MEASUREMENTS_MIN = 5
+    ckpts = find_all_checkpoints(PATH)
+    total_models = 0
+
+    # FIXME
+    #ckpts = ckpts[80:82]
+    for i, ckpt in enumerate(ckpts):
+
+        # FIXME -- infinite loop
+        if re.search("bert-base-cased-pyt", ckpt):
+            continue
+
+        # FIXME -- they don't use max_batch
+        if re.search("ncf", ckpt):
+            continue
+
+        print(f"{i+1} out of {len(ckpts)} checkpoints")
+        print(ckpt)
+
+        is_linear_inst = False
+        with open(ckpt) as f:
+            contents = f.read()
+            if re.search('instanceGroup": \[{"count": 3,', contents):
+                is_linear_inst = True
+        for model in get_models_in_checkpoint(ckpt):
+            total_models += 1
+            cmd = f"python3 /home/tgerdes/Code/model_analyzer/experiments/main.py --model-name {model} --generator QuickRunConfigGenerator --data-path {ckpt}"
+
+            if not is_linear_inst:
+                cmd = f"{cmd} --exponential-inst-count"
+            cmd_list = cmd.split(' ')
+            result = subprocess.run(cmd_list, stdout=subprocess.PIPE)
+            result_data = parse_experiment_output(result.stdout)
+
+            if 'Percentile' not in result_data:
+                print(f"{model} failed! Skipping")
+                continue
+            if 'Generator num configs' not in result_data:
+                print(f"{model} failed! Skipping")
+                continue
+
+            percentile = float(result_data['Percentile'])
+            num_measurement = int(result_data['Generator num configs'])
```

Comment:
Should this use "num measurements" and "missing num measurements" to get number of measurements?